HLTrigger/JetMET/interface/HLTHFJetCleaner.h

```diff
@@ -0,0 +1,224 @@
+      sigmaPhiPhiToken_(consumes<edm::ValueMap<float>>(iConfig.getParameter<edm::InputTag>("sigmaPhiPhi"))),
+      centralEtaStripSizeToken_(
+          consumes<edm::ValueMap<int>>(iConfig.getParameter<edm::InputTag>("centralEtaStripSize"))),
+      jetPtThresh_(iConfig.getParameter<double>("jetPtThresh")),
+      dphiJetMetThresh_(iConfig.getParameter<double>("dphiJetMetThresh")),
+      jetEtaMin_(iConfig.getParameter<double>("jetEtaMin")),
+      jetEtaMax_(iConfig.getParameter<double>("jetEtaMax")),
+      sigmaEtaPhiDiffThresh_(iConfig.getParameter<double>("sigmaEtaPhiDiffThresh")),
+      centralEtaStripSizeThresh_(iConfig.getParameter<int>("centralEtaStripSizeThresh")),
+      cornerCutSigmaEtaEta_(iConfig.getParameter<double>("cornerCutSigmaEtaEta")),
+      cornerCutSigmaPhiPhi_(iConfig.getParameter<double>("cornerCutSigmaPhiPhi")),
+      applyCornerCut_(iConfig.getParameter<bool>("applyCornerCut")),
+      applySigmaEtaPhiCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCut")),
+      applyStripSizeCut_(iConfig.getParameter<bool>("applyStripSizeCut")) {
+  produces<JetCollection>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+template <typename JetType>
+void HLTHFJetCleaner<JetType>::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  edm::Handle<std::vector<JetType>> jets;
+  edm::Handle<edm::View<reco::MET>> mets;
+
+  edm::Handle<edm::ValueMap<float>> sigmaEtaEtas;
+  edm::Handle<edm::ValueMap<float>> sigmaPhiPhis;
+  edm::Handle<edm::ValueMap<int>> centralEtaStripSizes;
+
+  iEvent.getByToken(jetToken_, jets);
+
+  iEvent.getByToken(sigmaEtaEtaToken_, sigmaEtaEtas);
+  iEvent.getByToken(sigmaPhiPhiToken_, sigmaPhiPhis);
+  iEvent.getByToken(centralEtaStripSizeToken_, centralEtaStripSizes);
+
+  // The set of cleaned jets
+  std::unique_ptr<JetCollection> cleaned_jets(new JetCollection);
+
+  // Loop over the jets and do cleaning
+  for (uint ijet = 0; ijet < jets->size(); ijet++) {
+    JetType jet = (*jets)[ijet];
+    bool withinEtaRange = (abs(jet.eta()) < jetEtaMax_) & (abs(jet.eta()) > jetEtaMin_);
+    // Cuts are applied to jets that are back to back with MET
+    iEvent.getByToken(metToken_, mets);
+    const float met_phi = mets->front().phi();
+    bool backToBackWithMet = deltaPhi(jet.phi(), met_phi) > dphiJetMetThresh_;
+    if (!backToBackWithMet) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // If the jet is outside the range that we're interested in,
+    // append it to the cleaned jets collection and continue
+    if ((jet.pt() < jetPtThresh_) | (!withinEtaRange)) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+
+    JetRef const jetRef(jets, ijet);
+
+    // Extract HF shape variables from the ValueMap inputs
+    float sigmaEtaEta = (*sigmaEtaEtas)[jetRef];
+    float sigmaPhiPhi = (*sigmaPhiPhis)[jetRef];
+    int centralEtaStripSize = (*centralEtaStripSizes)[jetRef];
```

Comment:
Use `auto const`. Also, looks like for VMap, handles are not needed (so they can be created via `iEvent.get`).